File: gempa.py
#!/usr/bin/env python3
# use python3

'''
Gempa - Info gempa from bmkg.go.id
This project was created by Dfv47 with Black Coder Crush. 
Copyright 22 - 12 - 2k19 @m_d4fv
'''

import requests, xml.dom.minidom as xmlget, os, time

# Color
a='\033[1;30m'
r='\033[1;31m'
g='\033[32;1m' 
y='\033[1;33m'
c='\033[1;36m' 
w='\033[1;37m' 
n='\033[0;00m' 

def banner():
    os.system('clear')
    print (y+"  _____  "+r+"{ "+a+"Python script "+r+"}                                   ")
    print (y+" |   __|___ _____ ___ ___  "+w+"Author "+r+": "+w+"M Daffa                  ")
    print (y+" |  |  | -_|     | . | .'| "+w+"Team   "+r+": "+w+"Black Coder Crush        ")
    print (y+" |_____|___|_|_|_|  _|__,| "+w+"Github "+r+": "+w+"https://github.com/md4fv ")
    print (y+"                 |_| "+r+"* "+a+"Check info gempa from bmkg.go.id           ")
    print (r+'\n Loading please wait..from bmkg.go.id..')
    time.sleep(3)

def main():
    response  = requests.get('http://data.bmkg.go.id/autogempa.xml')
    result    = response.text
    save      = open('result.xml', 'w')
    save.write('%s' % result)
    save.close()
    rxml      = xmlget.parse('result.xml')    
    tanggal   = rxml.getElementsByTagName('Tanggal')[0].firstChild.data
    jam       = rxml.getElementsByTagName('Jam')[0].firstChild.data
    lintang   = rxml.getElementsByTagName('Lintang')[0].firstChild.data
    bujur     = rxml.getElementsByTagName('Bujur')[0].firstChild.data
    magnitude = rxml.getElementsByTagName('Magnitude')[0].firstChild.data
    kedalaman = rxml.getElementsByTagName('Kedalaman')[0].firstChild.data
    wilayah   = rxml.getElementsByTagName('Wilayah1')[0].firstChild.data
    potensi   = rxml.getElementsByTagName('Potensi')[0].firstChild.data
    print (r+"\n { "+w+"Gempa Information "+r+"} ")
    print (y+' Tanggal     '+r+': '+y+tanggal+'')
    print (y+' Jam         '+r+': '+y+jam+'')
    print (y+' Lintang     '+r+': '+y+lintang+'')
    print (y+' Bujur       '+r+': '+y+bujur+'')
    print (y+' Magnitude   '+r+': '+y+magnitude+'')
    print (y+' Kedalaman   '+r+': '+y+kedalaman+'')
    print (y+' Wilayah     '+r+': '+y+wilayah+'')
    print (y+' Potensi     '+r+': '+y+potensi+'')


if __name__ == '__main__':
    banner()
    main()
    if os.path.exists('result.xml'):
        os.remove('result.xml')
    else:
        print (r+'The file does not exists')

File: gempa.php
<?php

error_reporting(0);

/*
@ Gempa - Info gempa from bmkg.go.id
@ This project was created by Dfv47 with Black Coder Crush. 
@ Copyright 25 - 12 - 2k19 @m_d4fv
@ Thank's to Akin
*/

class Gempa
{
	private $r = "\033[91m",
		$g = "\033[92m",
		$y = "\033[93m",
		$b = "\033[94m",
		$m = "\033[95m",
		$c = "\033[96m",
		$w = "\033[97m",
		$n = "\033[0m",
		$p = "\033[1;30m";
		
	public function __construct()
	{
		$this->banner();
		$this->printInfo();
	}
	
	// banner
	public function banner()
	{
		echo "{$this->y}  _____  {$this->r}{ {$this->p}PHP script {$this->r}}                                   \n";
		echo "{$this->y} |   __|___ _____ ___ ___  {$this->w}Author {$this->r}: {$this->w}M Daffa                  \n";
		echo "{$this->y} |  |  | -_|     | . | .'| {$this->w}Team   {$this->r}: {$this->w}Black Coder Crush        \n";
		echo "{$this->y} |_____|___|_|_|_|  _|__,| {$this->w}Github {$this->r}: {$this->w}https://github.com/md4fv \n";
		echo "{$this->y}                 |_| {$this->r}* {$this->p}Check info gempa from bmkg.go.id                \n";
	}

	// get info gempa
	private function printInfo()
	{
		$this->getInfo = file_get_contents("http://data.bmkg.go.id/lastgempadirasakan.xml");
		// parse xml
		$this->data = simplexml_load_string($this->getInfo) or die("error: cannot create object");
		// save variable
		$this->tanggal = $this->data->Gempa->Tanggal;
		$this->jam = $this->data->Gempa->Jam;
		$this->lintang = $this->data->Gempa->Lintang;
		$this->bujur = $this->data->Gempa->Bujur;
		$this->magnitude = $this->data->Gempa->Magnitude;
		$this->kedalaman = $this->data->Gempa->Kedalaman;
		$this->keterangan = $this->data->Gempa->Keterangan;
		$this->dirasakan = $this->data->Gempa->Dirasakan;
		// print info
		echo " {$this->r}{ {$this->w}Gempa Information{$this->r} }\n";
		echo " {$this->y}Tanggal     {$this->r}:{$this->y} {$this->tanggal} \n";
		echo " {$this->y}Jam         {$this->r}:{$this->y} {$this->jam} \n";
		echo " {$this->y}Lintang     {$this->r}:{$this->y} {$this->lintang} \n";
		echo " {$this->y}Bujur       {$this->r}:{$this->y} {$this->bujur} \n";
		echo " {$this->y}Magnitude   {$this->r}:{$this->y} {$this->magnitude} \n";
		echo " {$this->y}Kedalaman   {$this->r}:{$this->y} {$this->kedalaman} \n";
		echo " {$this->y}Keterangan  {$this->r}:{$this->y} {$this->keterangan} \n";
		echo " {$this->y}Dirasakan   {$this->r}:{$this->y} {$this->dirasakan} \n";
	}
}
$data = new Gempa();

